main.go

```diff
@@ -150,6 +150,8 @@ var (
 
 	dockerScan       = cli.Command("docker", "Scan Docker Image")
 	dockerScanImages = dockerScan.Flag("image", "Docker image to scan. Use the file:// prefix to point to a local tarball, otherwise a image registry is assumed.").Required().Strings()
+	dockerCache      = dockerScan.Flag("cache", "Use layer caching. Don't re-scan a layer that has already been scanned and is in the layer caching db.").Bool()
+	dockerCacheDB    = dockerScan.Flag("cache-db", "Path to the layer caching database. Default is trufflehog_layers.sqlite3").Default("trufflehog_layers.sqlite3").String()
```

Comment:
If we're persisting data, it may be prudent to make the name more generic in case there's other stuff worth caching in the future.

Reply:
Are you thinking about data caching outside of the Docker source or within it?

Reply:
I was thinking about data caching in general. I've previously experimented with using sqlite to do things like skip duplicate docker layers, binary files, and commits; cache GitHub API E-Tags; track progress so that scans can be resumed; etc. 

Then again, based on @rosecodym's comment about performance it may not be desirable to re-use the same database for multiple purposes.